subsys/usb/usb_transfer.c

```diff
@@ -49,7 +49,7 @@ static struct usb_transfer_data ut_data[CONFIG_USB_MAX_NUM_TRANSFERS];
 static struct usb_transfer_data *usb_ep_get_transfer(uint8_t ep)
 {
 	for (int i = 0; i < ARRAY_SIZE(ut_data); i++) {
-		if (ut_data[i].ep == ep) {
+		if (ut_data[i].ep == ep && ut_data[i].status != 0) {
```

Comment:
The whole thing around the transfer system is not designed to handle more than one transfer for specific endpoint. If an endpoint is used by two different contexts, then it should be checked with `usb_transfer_is_busy()`  before. I see API docu does not describe it. IMO, your change does not fix that and helps only if the transfers are not larger than endpoint MPS. I do not see any disadvantages of the patch yet, but commit message should be changed. I will block it for now.

Reply:
I see... maybe this check could be in `usb_transfer_sync()` to prevent starting of parallel transfers to the same endpoint. Caller can implement the waiting logic based on `usb_transfer_is_busy()`, but if is it not done, the deadlock is hard to find without explicit error message. If API is changed then `samples/net/wpanusb/src/wpanusb.c` shall be updated accordingly since it does parallel transfers to same endpoint.

Reply:
Updated commit with a check to prevent starting of the parallel transfers on same endpoint.